### PR TITLE
Generate docs for 7.12 pre-built rules

### LINKED ISSUE #565: [DOCS] Prebuilt rules for 7.12
## Description 

Generate docs for pre-built rules

## Acceptance Test Criteria

List all the ATC of each action and its intended result. 
As a user, when [action (e.g., viewing, clicking, selecting, etc.)] the [insert the expected result]. 
If the doc issue includes a procedure, number the steps in sequential order.  

## Notes

- Add the **"Team:Docs"** label to new issues. 
- Be sure to add the version number label. 
- Be sure to add any necessary screenshots for clarity. 
- Include any conditions or caveats that may affect customers.

### PR DESCRIPTION
**Pull request by brokensound77**
resolves #565 
dependent on #581 

Auto-generated docs with `python prebuilt-rules-scripts/generate.py generate "7.12.0" -l ../kibana-fork`

~[Preview link](https://security-docs_566.docs-preview.app.elstc.co/guide/en/security/master/prebuilt-rules.html).~

---

_*edit_

After we realized the changes from 7.11.2 were missing:
* the initial changes in this PR were reset to the last commit from master
* the 7.11.2 changes were added in #581
   - these changes were cherry picked into this branch/PR
* these docs were regenerated accounting for the changes

These should be accurate now.

For the deprecated rule, I manually modified them by:
* deleting the rule detail file
* adding a deprecated entry in the global rule changelog

**Updated preview [link](https://security-docs_566.docs-preview.app.elstc.co/guide/en/security/master/prebuilt-rules.html)**

### CODE REVIEW

File: docs/detections/prebuilt-rules/rule-details/azure-application-credential-modification.asciidoc
Comment (bm11100) [inline docs/detections/prebuilt-rules/rule-details/azure-application-credential-modification.asciidoc:36]: looks like these versions are getting bumped by 2 instead of 1

**Review by bm11100 — CHANGES_REQUESTED** [review]
Spot checked and most rule versions are being incremented by 2 instead of 1 somewhere in the code. Will need to track down where exactly this is. 

Example - 
image

**Review by nastasha-solomon — COMMENTED** [review]
Couldn't leave an in-line comment because wasn't sure where to enter it so I took a screenshot instead. I think the word "launched" might be misspelled in the title and description of the "Suspicious Hidden Child Process of Launchd" rule--both are highlighted to in the following image. Everything else looks good!

mispelling

File: docs/detections/prebuilt-rules/rule-details/bypass-uac-via-sdclt.asciidoc
Comment (threat-punter) [inline docs/detections/prebuilt-rules/rule-details/bypass-uac-via-sdclt.asciidoc:2]: Was this rule deprecated or something?

File: docs/detections/prebuilt-rules/rule-details/bash-shell-profile-modification.asciidoc
Comment (threat-punter) [inline docs/detections/prebuilt-rules/rule-details/bash-shell-profile-modification.asciidoc:61]: Looks like parts of the query are being rendered as comments in the code snippet.

image

File: docs/detections/prebuilt-rules/rule-details/component-object-model-hijacking.asciidoc
Comment (threat-punter) [inline docs/detections/prebuilt-rules/rule-details/component-object-model-hijacking.asciidoc:83]: Looks like the indices and look-back time was modified for this rule, but we're saying the update consists of formatting changes only. Is that intentional?

File: docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc
Comment (bm11100) [inline docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc:81]: how do we deal with deprecated rules? this one is deprecated but listed as an updated query. i don't think we yet have deprecation built into these docs.

File: docs/detections/prebuilt-rules/rule-details/execution-via-regsvcs-regasm.asciidoc
Comment (bm11100) [inline docs/detections/prebuilt-rules/rule-details/execution-via-regsvcs-regasm.asciidoc:35]: this one was also deprecated

File: docs/detections/prebuilt-rules/rule-details/component-object-model-hijacking.asciidoc
Comment (bm11100) [reply docs/detections/prebuilt-rules/rule-details/component-object-model-hijacking.asciidoc:83]: i believe if the rule query was not modified, then it'll go down as formatting only.

File: docs/detections/prebuilt-rules/rule-details/azure-application-credential-modification.asciidoc
Comment (brokensound77) [reply docs/detections/prebuilt-rules/rule-details/azure-application-credential-modification.asciidoc:36]: I found the source of the double bump. Every endpoint rule had `event.ingested` added in 7.11.2 [here](https://github.com/elastic/kibana/commit/df46bfcefdcfbba49e0c4842c3f6fc88822163df#diff-1d0a7eb78fc95721f1fa7f767d89590db57b4a041792076d1b68106cb8e5afab). 

So I will need to re-run all of this to include 7.11.2 and _then_ 7.12 to include those interim changes

File: docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc
Comment (brokensound77) [reply docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc:81]: > Couldn't leave an in-line comment because wasn't sure where to enter it so I took a screenshot instead. I think the word "launched" might be misspelled in the title and description of the "Suspicious Hidden Child Process of Launchd" rule--both are highlighted to in the following image. Everything else looks good!
> 
> mispelling

Thanks for finding this @nastasha-solomon. In this case, it is referring to a specific process names `launchd` so it is correct

**Comment by brokensound77** [comment]
There was a mass change to endpoint rules in 7.11.2 so that explains the double bump in some versions. In order to keep docs aligned, I created #581 to incorporate the changes for 7.11.2. Once that merges, I will cherry pick the changes here and regenerate these docs to fix the discrepancies.

File: docs/detections/prebuilt-rules/rule-details/bypass-uac-via-sdclt.asciidoc
Comment (brokensound77) [reply docs/detections/prebuilt-rules/rule-details/bypass-uac-via-sdclt.asciidoc:2]: yep. Looks like previously removed rules were handled like this:
image

Thought I think `deprecated` vs `removed` is better verbiage. Also, how deprecated rules are handled in Kibana is in flux and may change.

I will see how to incorporate this to these docs (even if just manually)

File: docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc
Comment (nastasha-solomon) [reply docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc:81]: Ah ok. Thanks for clarifying!

File: docs/detections/prebuilt-rules/rule-details/bash-shell-profile-modification.asciidoc
Comment (brokensound77) [reply docs/detections/prebuilt-rules/rule-details/bash-shell-profile-modification.asciidoc:61]: Nice find. I played with some of the other code escape blocks and understandably, this syntax causes issues with all of them. Most are interpreting the `/* ... */` as block comments, which is just KQL syntax. The only option at that point would be to do a generic escape with no syntax highlighting for these rules, but I think that is too much and creates inconsistency.

I think we are to leave as is until someone complains

File: docs/detections/prebuilt-rules/rule-details/execution-via-regsvcs-regasm.asciidoc
Comment (brokensound77) [reply docs/detections/prebuilt-rules/rule-details/execution-via-regsvcs-regasm.asciidoc:35]: This one was removed as of 7.13

File: docs/detections/prebuilt-rules/rule-details/bypass-uac-via-sdclt.asciidoc
Comment (brokensound77) [reply docs/detections/prebuilt-rules/rule-details/bypass-uac-via-sdclt.asciidoc:2]: see the edited note in the description

**Comment by brokensound77** [comment]
keeps failing with `WARNING: invalid reference: setgid-bit-set-via-chmod` and I cannot find the error anywhere

File: docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc
Comment (bm11100) [inline docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc:16]: This rule, along with the other deprecated one `Execution via Regsvcs/Regasm` are still appearing on the preview link. 

The below should be deleted -  
`docs/detections/prebuilt-rules/rule-details/setuid-setgid-bit-set-via-chmod.asciidoc`
`docs/detections/prebuilt-rules/rule-details/execution-via-regsvcs-regasm.asciidoc`

Comment (bm11100) [inline docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc:16]:
```suggestion
* setuid-setgid-bit-set-via-chmod
* execution-via-regsvcs-regasm
```

**Review by bm11100 — APPROVED** [review]
Just one comment on the two deprecated rules. The versioning looks good now. Going to approve as to not hold this up further.

**Review by jmikell821 — APPROVED** [review]

File: docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc
Comment (brokensound77) [reply docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc:16]: It is really confusing because we had several rules with very similar names/approaches.
`setuid-setgid-bit-set-via-chmod` is still a rule and different from the 7.12 deprecated rule `setgid-bit-set-via-chmod`


`execution-via-regsvcs-regasm` was not removed until after 7.12 and so will reflect in the next doc updates for 7.13

So I think we should be good on both of these

Comment (brokensound77) [reply docs/detections/prebuilt-rules/prebuilt-rules-changelog.asciidoc:16]: Should be ok as is: https://github.com/elastic/security-docs/pull/566#discussion_r603475100

**Pull request merged**
